### PR TITLE
Psl 218 convert pastel gtest

### PR DESCRIPTION
**Pull request by tanlemin91**

### CODE REVIEW

**Review by akobrin1 — CHANGES_REQUESTED** [review]
some files needs to be merged with the latest master branch (like interpreter.h|cpp).
Please rebase on current master branch.

File: src/gtest/test_sighash.cpp [hunk 1]
```diff
@@ -1,31 +1,35 @@
...
+#include <version.h>
+#include <sodium.h>
+#include <json_test_vectors.h>
 #include <univalue.h>
 
-extern UniValue read_json(const std::string& jsondata);
+using namespace std;
+using namespace testing;
+
+extern UniValue read_json(const string& jsondata);
```
[the remark this answers is not in the document]
Comment (tanlemin91) [reply src/gtest/test_sighash.cpp]: done thanks

File: src/script/interpreter.cpp [hunk 2]
```diff
@@ -13,10 +13,11 @@
 #include <pubkey.h>
 #include <script/script.h>
 #include <uint256.h>
-#include <enum_util.h>
```
[the remark this answers is not in the document]
Comment (tanlemin91) [reply src/script/interpreter.cpp]: done thanks

File: src/script/interpreter.cpp [hunk 3]
```diff
@@ -13,10 +13,11 @@
...
 
 using namespace std;
 
+typedef v_uint8 valtype;
```
[the remark this answers is not in the document]
Comment (tanlemin91) [reply src/script/interpreter.cpp]: done thanks

File: src/script/interpreter.h [hunk 4]
```diff
@@ -3,8 +3,15 @@
 // Copyright (c) 2009-2014 The Bitcoin Core developers
 // Copyright (c) 2018-2022 The Pastel Core developers
 // Distributed under the MIT software license, see the accompanying
-// file COPYING or https://www.opensource.org/licenses/mit-license.php.
-#include <climits>
+// file COPYING or http://www.opensource.org/licenses/mit-license.php.
+
+#ifndef BITCOIN_SCRIPT_INTERPRETER_H
```
[the remark this answers is not in the document]
Comment (tanlemin91) [reply src/script/interpreter.h]: done thanks

**Review by akobrin2 — APPROVED** [review]

**Review by akobrin1 — APPROVED** [review]

**Pull request merged**